pilot/pkg/model/sidecar.go

```diff
@@ -980,7 +980,7 @@ func (sc *SidecarScope) appendSidecarServices(servicesAdded map[host.Name]sideca
 		// Update index as well, so that future reads will merge into the new service
 		foundSvc.svc = copied
 		servicesAdded[foundSvc.svc.Hostname] = foundSvc
-		sc.servicesByHostname[s.Hostname] = s
+		sc.servicesByHostname[s.Hostname] = copied
```

Comment:
I fixed in https://github.com/istio/istio/pull/50691, and also added a test for this

Reply:
Oh nice, missed that! Thanks